src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingSourceGenerator.Helpers.cs

```diff
@@ -85,30 +92,21 @@ private static class NotSupportedReason
 
         private static class TypeFullName
         {
-            public const string Array = "System.Array";
             public const string ConfigurationKeyNameAttribute = "Microsoft.Extensions.Configuration.ConfigurationKeyNameAttribute";
             public const string Dictionary = "System.Collections.Generic.Dictionary`2";
             public const string GenericIDictionary = "System.Collections.Generic.IDictionary`2";
             public const string HashSet = "System.Collections.Generic.HashSet`1";
-            public const string ISet = "System.Collections.Generic.ISet`1";
-            public const string IConfigurationSection = "Microsoft.Extensions.Configuration.IConfigurationSection";
             public const string IConfiguration = "Microsoft.Extensions.Configuration.IConfiguration";
+            public const string IConfigurationSection = "Microsoft.Extensions.Configuration.IConfigurationSection";
             public const string IDictionary = "System.Collections.Generic.IDictionary";
+            public const string ISet = "System.Collections.Generic.ISet`1";
             public const string IServiceCollection = "Microsoft.Extensions.DependencyInjection.IServiceCollection";
             public const string List = "System.Collections.Generic.List`1";
         }
 
         private static bool TypesAreEqual(ITypeSymbol first, ITypeSymbol second)
                 => first.Equals(second, SymbolEqualityComparer.Default);
 
-        private enum InitializationKind
-        {
-            None = 0,
-            SimpleAssignment = 1,
-            AssignmentWithNullCheck = 2,
-            Declaration = 3,
-        }
-
         private sealed class SourceWriter
```

Comment:
I wonder why we need this class at all. The Regex source generator just uses an IntendedTextWriter:

https://github.com/dotnet/runtime/blob/967250cc86b4c613015066b47b882e7424fdb590/src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.cs#L120-L122

Why can't we follow the same pattern here?

Reply:
Sure I'll look into switching to this.

Reply:
It looks like the Regex source generator doesn't use multi-line strings:

https://github.com/dotnet/runtime/blob/967250cc86b4c613015066b47b882e7424fdb590/src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs#L210-L218

When using a `CodeWriter` (which derives from `IndentedTextWriter`) in the RequestDelegateGenerator, it looks like we don't use multi-line strings either:

https://github.com/dotnet/aspnetcore/blob/3265dc6a9b05c74b199aa39351e5413317df9ad5/src/Http/Http.Extensions/gen/RequestDelegateGenerator.cs#L106-L127

However, we do use multi-line strings in the RequestDelegateGenerator when just appending strings together

https://github.com/dotnet/aspnetcore/blob/3265dc6a9b05c74b199aa39351e5413317df9ad5/src/Http/Http.Extensions/gen/RequestDelegateGeneratorSources.cs#L22-L40

cc @captainsafia @stephentoub - thoughts on how to generate code here?

Reply:
Maybe, but I'd say we should use a component that does not prohibit use of multi-line strings in future iterations. I find that it's easier to write and understand in practice:

https://github.com/eiriktsarpalis/typeshape-csharp/blob/dcb77f0c94ca762c4c98a6885df117fa4ddcd2a1/src/TypeShape.SourceGenerator/SourceFormatter/SourceFormatter.Constructors.cs#L26-L37

Perhaps having a bespoke class in the `Common` folder that does support it in all the right ways this might be appropriate.

Reply:
I think that being able to write both single and multi-line strings conveniently is beneficial and we should permit source generators to do both. Multi-line strings make the code more readable while single lines help with source composition/nesting.

Reply:
We generally use multiline strings in the RequestDelegateGenerator if the code emitted doesn't depend on any conditions and doesn't have complex indentation requirements. As @eiriktsarpalis mentioned, multi-line strings are a lot harder to read then invocations to the `CodeWriter`.